src/util/dynamicfilesequencedownloader.cpp

```diff
@@ -0,0 +1,657 @@
+    requestDataInfo(httpInfoRequest);
+    std::string httpDataRequest = buildDataHttpRequest(
+        _dataMinTime,
+        _dataMaxTime,
+        _dataID,
+        _dataURL
+    );
+    requestAvailableFiles(httpDataRequest, _syncDir);
+}
+
+void DynamicFileSequenceDownloader::deinitialize(bool cacheFiles) {
+    std::vector<File*>& currentlyDownloadingFiles = filesCurrentlyDownloading();
+    for (File* file : currentlyDownloadingFiles) {
+        file->download->cancel();
+        file->download->wait();
+        std::error_code ec;
+        std::filesystem::remove(file->path, ec);
+        if (ec) {
+            LERROR(std::format("Failed to delete unfinished file: {}", file->path));
+        }
+        else {
+            LINFO(std::format("Removing unfinished download:: {}", file->path));
+        }
+    }
+    if (!cacheFiles) {
+        if (!std::filesystem::exists(_syncDir)) {
+            return;
+        }
+        for (auto& file : std::filesystem::directory_iterator(_syncDir)) {
+            std::error_code ec;
+            std::filesystem::remove(file.path(), ec);
+            if (ec) {
+                LERROR(std::format("Failed to delete file: {}", file.path()));
+            }
+        }
+    }
+    if (std::filesystem::is_empty(_syncDir)) {
+        std::filesystem::remove(_syncDir);
+    }
+}
+
+void DynamicFileSequenceDownloader::requestDataInfo(std::string httpInfoRequest) {
+    HttpMemoryDownload response = HttpMemoryDownload(httpInfoRequest);
+    response.start();
+    response.wait();
+
+    bool success = false;
+    int attempt = 0;
+    constexpr int maxRetries = 1;
+
+    /********************
+    *   Example response
+    *********************
+    *{
+    *    "availability": {
+    *        "startDate": "2017-07-01T00:42:02.0Z",
+    *        "stopDate" : "2017-09-30T22:43:18.0Z"
+    *    },
+    *        "description" : "WSA 4.4 field line trace from the SCS outer boundary to the
+    *                         source surface",
+    *        "id" : 1177
+    *}
+    ********************/
+    while (attempt <= maxRetries && !success) {
+        try {
+            std::vector<char> responseText = response.downloadedData();
+            if (responseText.empty()) {
+                throw std::runtime_error("Empty HTTP response");
+            }
+            nlohmann::json jsonResult = nlohmann::json::parse(responseText);
+            success = true;
+            _dataMinTime = Time::convertTime(
+                jsonResult["availability"]["startDate"].get<std::string>()
+            );
+            _dataMaxTime = Time::convertTime(
+                jsonResult["availability"]["stopDate"].get<std::string>()
+            );
+        }
+        catch (const nlohmann::json::parse_error& e) {
+            LWARNING(std::format("JSON parse error: {}", e.what()));
+        }
+        catch (const std::exception& e) {
+            LWARNING(std::format("HTTP or other error: {}", e.what()));
+        }
```

Comment:
```suggestion
        catch (const ghoul::RuntimeError& e) {
            LWARNING(std::format("HTTP or other error: {}", e.what()));
        }
```

I guess you mean this?